Control/move_to_pose/move_to_pose.py

```diff
@@ -63,6 +67,12 @@ def move_to_pose(x_start, y_start, theta_start, x_goal, y_goal, theta_goal):
         if alpha > np.pi / 2 or alpha < -np.pi / 2:
             v = -v
 
+        if abs(v) > _MAX_LINEAR_SPEED:
+            v = v/abs(v) * _MAX_LINEAR_SPEED 
```

Comment:
Why you don’t just set maximum speed?

Reply:
We don't set the speed to max speed since we want to keep the control law alive. Note that when the robot gets to target (or very close to it) the robot speed should approaches zero.

Reply:
Sorry I don’t understand it well… Could elaborate it?
v is velocity, so it should be under the specification, right?
> Note that when the robot gets to target (or very close to it) the robot speed should approaches zero.

Yes. In that case, the speed must be under specification, your code does not work, right?

Reply:
Sure. The actual name of 'v' should have been 'command_linear_velocity' which is the velocity the controller commands the robot to move. When far from the target, rho and thus v are high. Since in reality any robot has a max speed, we need to limit the 'command_linear_velocity' to the robot's max speed. How to categorize the variables (robot specification vs controller parameters) really depends on how we look at the system. I would separate controller and plant (i.e., the robot) as different subsystem modules and therefore It is better to keep v as a controller parameter than the robot specification.

Reply:
OK. Thank you. I understood. Could you please use ‘np.sign’ instead of ‘v/-abs(v)’?

Reply:
You are welcome. Yes, that is a good idea. I just updated to np.sign().